subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/GenerateMavenPom.java

```diff
@@ -137,29 +137,32 @@ private MavenPomFileGenerator.MavenPomSpec computeMavenPomSpec() {
             pomInternal.writeGradleMetadataMarker());
         pomGenerator.configureFrom(pomInternal);
 
-        for (MavenDependency mavenDependency : pomInternal.getApiDependencyManagement()) {
-            pomGenerator.addApiDependencyManagement(mavenDependency);
+        MavenPomDependencies dependencies = pomInternal.getDependencies().get();
+        for (MavenDependencyInternal mavenDependency : dependencies.getApiDependencyManagement()) {
+            pomGenerator.addDependencyManagement(mavenDependency, "compile");
         }
 
-        for (MavenDependency mavenDependency : pomInternal.getRuntimeDependencyManagement()) {
-            pomGenerator.addRuntimeDependencyManagement(mavenDependency);
```

Comment:
💭 I think I actually prefer the way the old code reads here, with the named methods, rather than the scope string arguments.  Other opinions may vary.

Reply:
Reverted to named methods